### PR TITLE
[DNM] Added support for custom contentViewFactory

### PR DESCRIPTION
**Pull request by Lizards**
I have a RequireJS project that includes several Livefyre SDK applications from source, not using the Livefyre.js loader for anything except Auth and fyre.conv.  I disabled the modal from Livefyre.js's automatic spawning of streamhub-permalink so that I can include streamhub-permalink in my source and customize its behavior.  All I really want to do at this point is use a custom ContentView in the permalink modal.

I saw the 'opts.baseFactory' in permalink-view-factory, but no way to actually pass in that option, so that's what I added here.  So in my app I can do:

``` javascript
   new Permalink({
        contentViewFactory: new MyContentViewFactory()
    });
```

### CODE REVIEW

File: src/default-permalink-content-renderer.js [hunk 1]
```diff
@@ -6,9 +6,10 @@ var styles = require('css!./css/styles.css');
 var sdkStyles = require('css!streamhub-sdk/css/style.css');
 var packageAttribute = require('./package-attribute');
 
-var defaultPermalinkContentHandler = function (content) {
+var defaultPermalinkContentHandler = function (content, opts) {
+    opts = opts || {};
     //Get the view for the content
-    var cvf = permalinkViewFactory(),
```
Comment (gobengo) [inline src/default-permalink-content-renderer.js]: Passing it into `permalinkViewFactory` will mixin [some behavior](https://github.com/Lizards/streamhub-permalink/blob/support_contentViewFactory_param/src/permalink-view-factory.js#L26) that you (or someone else using this) may not necessarily want if they're trying to pass a completely custom contentViewFactory.

How about this line as
```javascript
var cvs = opts.contentViewFactory || permalinkViewFactory();
```

Comment (Lizards) [inline src/default-permalink-content-renderer.js]: Still works for me, the presence of opts.baseFactory in permalinkViewFactory threw me off, thinking the intention was to always apply certain behavior to views used with this permalink app.  Branch updated.

Comment (gobengo) [inline src/default-permalink-content-renderer.js]: A reasonable throw-off. I didn't write the baseFactory thing :P

**Comment by gobengo** [comment]
I think we intended to allow for something like this:
```javascript
var ADIPermalink = new Permalink();
ADIPermalink.preventDefault('content');
ADIPermalink.on('content', function () {
  // If you want to use a slight deviation of the default, you can still call it
  // (but if you wanted something completely different, you could do something else)
  require('streamhub-permalink/default-permalink-content-renderer')(this.get('content'), {
    contentViewFactory: MyContentViewFactory();
  });
});
```

**Pull request closed without merging**